src/relay/qnn/util.h

```diff
@@ -62,19 +52,15 @@ static inline bool IsValidOpInputType(const QuantizeOpType& op_type,
 static inline bool IsValidOpOutputType(const QuantizeOpType& op_type,
         const DataType& out_dtype) {
   switch (op_type) {
-    case QuantizeOpType::Quantize:
-      return IsQNNDataType(out_dtype);
-    case QuantizeOpType::Dequantize:
-      return out_dtype == Float(32);
     case QuantizeOpType::Requantize:
-      return IsQNNDataType(out_dtype);
+      return out_dtype == Int(8) || out_dtype == UInt(8) || out_dtype == Int(32);
     default:
       return false;
   }
 }
 
 static inline const int32_t GetQmin(const DataType& dtype) {
-  CHECK(IsQNNDataType(dtype))
+  CHECK_LE(dtype.bits(), 32)
```

Comment:
The types are discrete, but the Check here is CHECK_LE which means that technically this allows 16 bit values ?

Reply:
Yes this is equivalent to num_of_bits <= 32